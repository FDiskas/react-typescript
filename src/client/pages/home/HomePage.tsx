import * as React from 'react';
import * as styles from './HomePage.scss';

export interface IHomePageProps {}

export interface IHomePageState {}

export class HomePage extends React.Component<IHomePageProps, IHomePageState> {
  constructor(props: IHomePageProps) {
    super(props);
  }

  render() {
    return <main className={styles.page}>Component name: HomePage</main>;
  }
}
