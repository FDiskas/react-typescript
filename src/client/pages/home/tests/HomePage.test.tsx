import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { HomePage } from '../HomePage';

describe('HomePage.js', () => {
    it('renders without crashing', () => {
        shallow(<HomePage/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<HomePage/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
