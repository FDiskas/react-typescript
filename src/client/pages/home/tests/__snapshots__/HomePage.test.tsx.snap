// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage.js matches snapshot 1`] = `
<main
  className="page"
>
  Component name: HomePage
</main>
`;
